back/src/users/resolvers/mutations/editProfile.ts

```diff
@@ -19,16 +19,16 @@ export async function editProfileFn(
   payload: MutationEditProfileArgs
 ) {
   const editProfileSchema = yup.object({
-    name: yup.string().required().isSafeSSTI(),
+    name: yup.string().isSafeSSTI(),
     phone: yup.string()
   });
   editProfileSchema.validateSync(payload);
 
   const { name, phone } = payload;
 
-  const data = {
-    ...(name !== undefined ? { name } : {}),
-    ...(phone !== undefined ? { phone } : {})
+  const data: { name?: string; phone?: string } = {
+    ...(name !== undefined ? { name: name as string } : {}),
+    ...(phone !== undefined ? { phone: phone as string } : {})
```

Comment:
Le téléphone peut être vidé via null mais pas le nom. Pour un typing un peu plus précis et éviter les `as` on pourrait faire ça je pense:
```suggestion
  const data: { name?: string; phone?: string | null } = {
    ...(name != null ? { name } : {}),
    ...(phone !== undefined ? { phone } : {})
  };
```